File: code_examples/3-Structured_Problem_Solving/16-Break/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a break statement

#include <stdio.h>

int main() {
    // this function uses a break statement
    int positiveInteger;

    // input
    printf("Enter how many times to repeat: ");
    scanf("%d", &positiveInteger);
    printf("\n");

    // process & output
    for (int loopCounter = 0; loopCounter < positiveInteger; loopCounter++) {
        printf("%d time(s) through the loop.\n", loopCounter);
        if (loopCounter >= 5) {
            break;
        }
    }

    printf("\nDone.\n");
    return 0;
}

File: code_examples/3-Structured_Problem_Solving/16-Break/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a break statement
*/

using System;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function uses a break statement

        int positiveInteger;

        // input
        Console.Write("Enter how many times to repeat: ");
        positiveInteger = Convert.ToInt32(Console.ReadLine());
        Console.WriteLine();

        // process & output
        for (int loopCounter = 0; loopCounter < positiveInteger; loopCounter++) {
            Console.WriteLine(loopCounter + " time(s) through the loop.");
            if (loopCounter >= 5) {
                break;
            }
        }

        Console.WriteLine("\nDone.");
    }
}